newswires/app/db/FingerpostWireEntry.scala

```diff
@@ -71,26 +71,35 @@ object FingerpostWireEntry extends SQLSyntaxSupport[FingerpostWireEntry] {
   }
 
   case class QueryResponse(
-      results: List[FingerpostWireEntry]
+      results: List[FingerpostWireEntry],
+      keywordCounts: Map[String, Int]
   )
   private object QueryResponse {
     implicit val writes: OWrites[QueryResponse] = Json.writes[QueryResponse]
   }
 
   def query(
       maybeFreeTextQuery: Option[String],
+      maybeKeywords: Option[List[String]],
       page: Int = 0,
       pageSize: Int = 250
   ): QueryResponse = DB readOnly { implicit session =>
     val effectivePage = clamp(0, page, 10)
     val effectivePageSize = clamp(0, pageSize, 250)
     val position = effectivePage * effectivePageSize
 
-    val whereClause = List(
+    val whereClauses = List(
+      maybeKeywords.map(keywords =>
+        sqls"""(${FingerpostWireEntry.syn.column(
+            "content"
+          )} -> 'keywords') @> ${Json.toJson(keywords).toString()}::jsonb"""
```

Comment:
not a bad shout, although not sure how I'd do this, but also since the keywords will be coming from the `/api/keywords` endpoint, I suspect it won't be an issue in practice